ngraph/frontend/generic/include/frontend_manager/frontend_manager.hpp

```diff
@@ -312,11 +312,19 @@ enum FrontEndCapabilities {
 class NGRAPH_API FrontEndManager
 {
 public:
-    FrontEndManager () {}
-    FrontEnd::Ptr loadByFramework (const std::string& framework, FrontEndCapabilities fec = FEC_DEFAULT);
-    FrontEnd::Ptr loadByModel (const std::string& path, FrontEndCapabilities fec = FEC_DEFAULT);
-    std::vector<std::string> availableFrontEnds () const;
+    FrontEndManager();
+    ~FrontEndManager();
+    FrontEnd::Ptr loadByFramework(const std::string& framework, FrontEndCapabilities fec = FEC_DEFAULT);
+    FrontEnd::Ptr loadByModel(const std::string& path, FrontEndCapabilities fec = FEC_DEFAULT);
+    std::vector<std::string> availableFrontEnds() const;
+
+    using FrontEndFactory = std::function<FrontEnd::Ptr(FrontEndCapabilities fec)>;
+    void registerFrontEnd(const std::string& name, FrontEndFactory creator);
```

Comment:
We have to implement logic that loads frontends as shared libraries like Reader API does. Created ticket 52321 for you.

Reply:
What is the purpose for user to register FE in FE Manger? Just to have ability to instantiate it in loadByFramework? Or to hide probing logic to use loadByModel?

Reply:
This is to have code more generic, not dependent on actual frontends.
Regarding plugin API - it will be good. But API like registerFrontEnd in code will also be very useful, at least for mocking testing